Code/src/main/java/nl/utwente/viskell/haskell/env/FunctionInfo.java

```diff
@@ -1,14 +1,28 @@
 package nl.utwente.viskell.haskell.env;
 
 import nl.utwente.viskell.haskell.type.Type;
+import nl.utwente.viskell.ui.serialize.Bundleable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 public abstract class FunctionInfo {
 
+    private static final Map<String, String> subClassMap;
+    static {
+        Map<String, String> aMap = new HashMap<>();
+        aMap.put(CatalogFunction.class.getSimpleName(), CatalogFunction.class.getName());
+        subClassMap = Collections.unmodifiableMap(aMap);
```

Comment:
I haven't tested it, but this could perhaps be an `ImmutableMap.of` which would be slightly shorter.